src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDeclareVisibleInstanceFields.cs

```diff
@@ -24,15 +25,25 @@ public abstract class DoNotDeclareVisibleInstanceFieldsAnalyzer : DiagnosticAnal
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182141.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterSymbolAction(obj =>
+            {
+                var field = (IFieldSymbol)obj.Symbol;
+
+                // Only report diagnostic on non-static, non-const, public fields
+                if (!field.IsStatic && !field.IsConst && field.IsPublic())
```

Comment:
Where's the FxCop implementation - I can't seem to find it on http://referencesource.microsoft.com